handlers/mapping_handlers.go

```diff
@@ -0,0 +1,404 @@
+}
+
+// GetMappingHandler gets a single user from a conversation
+func (env *Env) GetMappingHandler(w http.ResponseWriter, r *http.Request) {
+	userID, err := strconv.ParseInt(r.Header.Get("User-ID"), 10, 64)
+	if err != nil {
+		errMsg := "Invalid user ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	vars := mux.Vars(r)
+	conversationID, err := strconv.ParseInt(vars["conversation_id"], 10, 64)
+	if err != nil {
+		errMsg := "Invalid conversation ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+	memberID, err := strconv.ParseInt(vars["user_id"], 10, 64)
+	if err != nil {
+		errMsg := "Invalid user ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	conversation, err := env.getConversation(w, conversationID)
+	if err != nil || conversation == nil {
+		return
+	}
+
+	sessionMember, err := env.DB.GetUserConversationMapping(userID, conversationID)
+	if err != nil {
+		internalServerError(w, err)
+		return
+	}
+	if sessionMember == nil {
+		errMsg := fmt.Sprintf("User %d is not in conversation %d", userID, conversationID)
+		log.Println(errMsg)
+		http.Error(w, "Conversation not found", http.StatusNotFound)
+		return
+	}
+
+	member, err := env.DB.GetUserConversationMapping(memberID, conversationID)
+	if err != nil {
+		internalServerError(w, err)
+		return
+	}
+	if member == nil {
+		errMsg := fmt.Sprintf("User %d is not in conversation %d", userID, conversationID)
+		log.Println(errMsg)
+		http.Error(w, "User not found", http.StatusNotFound)
+		return
+	}
+
+	w.Header().Add("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(member)
+}
+
+// GetMappingsHandler gets all users from a conversation
+func (env *Env) GetMappingsHandler(w http.ResponseWriter, r *http.Request) {
+	userID, err := strconv.ParseInt(r.Header.Get("User-ID"), 10, 64)
+	if err != nil {
+		errMsg := "Invalid user ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	vars := mux.Vars(r)
+	conversationID, err := strconv.ParseInt(vars["conversation_id"], 10, 64)
+	if err != nil {
+		errMsg := "Invalid conversation ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	sessionMember, err := env.DB.GetUserConversationMapping(userID, conversationID)
+	if err != nil {
+		internalServerError(w, err)
+		return
+	}
+	if sessionMember == nil {
+		errMsg := fmt.Sprintf("User %d is not in conversation %d", userID, conversationID)
+		log.Println(errMsg)
+		http.Error(w, "Conversation not found", http.StatusNotFound)
+		return
+	}
+
+	members, err := env.DB.GetUserConversationMappings(conversationID)
+	if err != nil {
+		internalServerError(w, err)
+		return
+	}
+	memberList := &models.UserConversationMappingList{Users: members}
+
+	// TODO: Get user names from karen?
```

Comment:
Yeah, forgot to elaborate on this. The question is if we want to do sorting on the front-end or back-end, because if we want the back-end to sort the list of users by something useful, we'd need to get that from `karen`.

Reply:
Oh true. Hmmm. I guess we'd probably be doing the same thing in either case, i.e. client makes the request to `ether`, `ether` gets the sorted users from `karen`, then `ether` returns the user-conversation mapping merged with the sorted users or client makes request to `ether` gets the user-conversation mapping, makes request to `karen` get the sorted users and then matches the mapping to the sorted users itself. I think it'd be better to do it in the front-end, so that we don't have to send a request to `ether` and `karen` everytime the sorting option changes.